### PR TITLE
Closes #4: Build add to cart, cart PORO, cart_controller

### LINKED ISSUE #4: 4. Adding items to the cart
Background: Items, and a user that is not logged in

As a visitor When I visit any page with an item on it I should see a link or button for “Add to Cart” When I click “Add to cart” for that item And I click a link or button to view cart And my current path should be “/cart” And I should see a small image, title, description and price for the item I just added And there should be a “total” price for the cart that should be the sum of all items in the cart

### PR DESCRIPTION
**Pull request by Aram-Anderson**
#### @mentions of the person or team responsible for reviewing proposed changes:
@dphilla @mikeyduece @liambarstad

### CODE REVIEW

File: app/views/items/index.html.erb [hunk 1]
```diff
@@ -10,10 +10,10 @@
         <%= image_tag item.image, class: "activator", id: "card-image" %>
       </div>
       <div class="card-content">
-        <span class="card-title activator grey-text text-darken-4"><%= link_to item.name.capitalize, item_path(item) %><i class="material-icons right">more_vert</i></span>
+        <span class="card-title activator grey-text text-darken-4"><%= link_to item.name.capitalize, item_path(item) %><br><%= link_to "Add to Cart", cart_path(item_id: item.id), class: "waves-effect waves-light btn", method: 'post' %><p><%= item.price.round(2) %></p><i class="material-icons right">more_vert</i></span>
```
Comment (mikeheft) [inline app/views/items/index.html.erb:13]: Good catch on the passing specific params for the session!

**Review by mikeheft — APPROVED** [review]
Nice work! Really like the passing specific params to the path. Loving the ease of materialize for the styling!
👍 ⚡️

File: app/models/cart.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  def add_item(item_id)
+    contents[item_id.to_s] += 1
+  end
+
+  def total_count
+    contents.values.sum
+  end
+
+  def total_cost
```
Comment (dphilla) [inline app/models/cart.rb:21]: Nice work here with the total cost

**Review by dphilla — APPROVED** [review]
Excellent werk.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,3 +1,8 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_action :set_cart
+
+  def set_cart
```
Comment (liambarstad) [inline app/controllers/application_controller.rb:5]: right on! will use this method throughout

File: spec/features/items/user_can_add_an_item_to_cart_spec.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    end
+  end
+  context "and adds items to the cart" do
+    scenario "visits cart and sees items" do
+      item1 = create(:item)
+      item2 = (:item)
+
+      visit items_path
+
+      first(:link, "Add to Cart").click
```
Comment (liambarstad) [inline spec/features/items/user_can_add_an_item_to_cart_spec.rb:21]: cool wording

**Review by liambarstad — APPROVED** [review]

**Pull request merged**